lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -10500,6 +10501,115 @@ void EmitYieldStar(ParseNodeUni* yieldStarNode, ByteCodeGenerator* byteCodeGener
+    pnodeClass->pnodeConstructor->location = pnodeClass->location;
+    Js::RegSlot protoLoc = funcInfo->AcquireTmpRegister();
+    BeginEmitBlock(pnodeClass->pnodeBlock, byteCodeGenerator, funcInfo);
+
+    Js::RegSlot frameDisplayLoc = funcInfo->frameDisplayRegister != Js::Constants::NoRegister ? funcInfo->frameDisplayRegister : funcInfo->GetEnvRegister();
+    Js::RegSlot tmpFrameDisplayLoc = funcInfo->AcquireTmpRegister();
+    frameDisplayLoc = byteCodeGenerator->PrependLocalScopes(frameDisplayLoc, tmpFrameDisplayLoc, funcInfo);
+    if (frameDisplayLoc != tmpFrameDisplayLoc)
+    {
+        funcInfo->ReleaseTmpRegister(tmpFrameDisplayLoc);
+    }
+
+    // Extends
+    if (pnodeClass->pnodeExtends)
+    {
+        Emit(pnodeClass->pnodeExtends, byteCodeGenerator, funcInfo, false);
+
+        byteCodeGenerator->StartStatement(pnodeClass->pnodeExtends);
+
+        Js::RegSlot ctorParentLoc = funcInfo->IsTmpReg(pnodeClass->pnodeExtends->location) ? pnodeClass->pnodeExtends->location : funcInfo->AcquireTmpRegister();
+        Js::RegSlot protoParentLoc = funcInfo->AcquireTmpRegister();
+
+        Js::ByteCodeLabel labelParentsFound = byteCodeGenerator->Writer()->DefineLabel();
+        byteCodeGenerator->Writer()->BrReg3(Js::OpCode::CheckExtends, labelParentsFound, ctorParentLoc, protoParentLoc, pnodeClass->pnodeExtends->location);
+        if (pnodeClass->pnodeExtends->location != ctorParentLoc)
+        {
+            byteCodeGenerator->Writer()->Reg2(Js::OpCode::Ld_A_ReuseLoc, ctorParentLoc, pnodeClass->pnodeExtends->location);
+        }
+        uint cacheId = funcInfo->FindOrAddInlineCacheId(ctorParentLoc, Js::PropertyIds::prototype, false, false);
+        byteCodeGenerator->Writer()->PatchableProperty(Js::OpCode::LdFld_ReuseLoc, protoParentLoc, ctorParentLoc, cacheId);
+        byteCodeGenerator->Writer()->BrReg1(Js::OpCode::BrOnObjectOrNull_A, labelParentsFound, protoParentLoc);
+        byteCodeGenerator->Writer()->W1(Js::OpCode::RuntimeTypeError, SCODE_CODE(JSERR_InvalidPrototype));
+        byteCodeGenerator->Writer()->MarkLabel(labelParentsFound);
+        if (frameDisplayLoc == funcInfo->frameDisplayRegister || frameDisplayLoc  == funcInfo->GetEnvRegister())
+        {
+            byteCodeGenerator->Writer()->Reg4U(Js::OpCode::InitClass, pnodeClass->location, protoLoc, ctorParentLoc, protoParentLoc, pnodeClass->pnodeConstructor->nestedIndex);
+        }
+        else
+        {
+            byteCodeGenerator->Writer()->Reg5U(Js::OpCode::InitInnerClass, pnodeClass->location, protoLoc, ctorParentLoc, protoParentLoc, frameDisplayLoc, pnodeClass->pnodeConstructor->nestedIndex);
+        }
+
+        funcInfo->ReleaseTmpRegister(protoParentLoc);
+        funcInfo->ReleaseTmpRegister(ctorParentLoc);
+
+        byteCodeGenerator->EndStatement(pnodeClass->pnodeExtends);
+    }
+    else
+    {
+        if (frameDisplayLoc == funcInfo->frameDisplayRegister || frameDisplayLoc  == funcInfo->GetEnvRegister())
```

Comment:
Can you describe the difference between `InitBaseClass` and `InitInnerBaseClass` used here?

Reply:
"Inner" refers to the fact that the class is defined in an inner or lexical scope. So we need to encode a byte code register that's holding the frame display we created for the scope. Otherwise, the frame display register is implied.